manim/scene/scene_file_writer.py

```diff
@@ -40,6 +40,55 @@
     from manim.renderer.opengl_renderer import OpenGLRenderer
 
 
+def to_av_frame_rate(fps):
+    epsilon1 = 1e-4
+    epsilon2 = 0.02
```

Comment:
Out of curiosity, where did these numbers come from?

Reply:
Maybe I should have added a comment about these epsilons.
pyav wants fractions but at this point in the code all we have is a float. Though, strictly speaking, IEEE 754 floats are fractions, converting from floats to meaningful fractions is not obvious. I say meaningful because floats can only represent powers of two so something like (50 * 2 / 3) does not have an exact representation in binary. For (50 * 2 / 3), `Fraction.from_float` does not return `Fraction(100, 3)`, as one would naïvely expect:

```
>>> from fractions import Fraction
>>> Fraction.from_float(50 * 2 / 3)
Fraction(4691249611844267, 140737488355328)
```

If a frame rate is obtained via some calculation, implicit float rounding rules may result in frame rate values like 49.999... instead of 50. `Fraction.from_float` returns:

```
>>> Fraction(50.0 - 2**-47)
Fraction(7036874417766399, 140737488355328)
```

Using this value rather than `Fraction(50, 1)` could lead to subtle issues.

I don't think this problem may be solved in general. I made the assumption that frame rates would be either whole values or multiples of 1000 / 1001 (because NTSC) specified with two decimals.
I chose epsilon1 and epsilon2 such that:

- the epsilon1 check accepts 59.99999... as 60 but rejects 59.94
- the epsilon2 check accepts common multiples of (1000/1001) 

If the need to support other fractions arises, this function will have to be revisited.